### PR TITLE
Added proof dec without Merlin

### PR DESCRIPTION
**Pull request by iquerejeta**
Current function depends on the ZKP toolkit, which uses the Merlin crate to create the Transcripts. The latter is hard to mimic in solidity, and hence I added a function that does not depend on Merlin

### CODE REVIEW

**Review by gpestana — APPROVED** [review]
LGMT!

File: src/private.rs [hunk 1]
```diff
@@ -110,6 +111,43 @@ impl SecretKey {
...
+        &self,
+        ciphertext: &Ciphertext,
+        message: &RistrettoPoint,
+    ) -> ((CompressedRistretto, CompressedRistretto), Scalar) {
+        let pk = PublicKey::from(self);
+        let announcement_random = Scalar::random(&mut OsRng);
+        let announcement_base_G = announcement_random * RISTRETTO_BASEPOINT_POINT;
+        let announcement_base_ctxtp0 = announcement_random * ciphertext.points.0;
+
+        let challenge = Scalar::from_hash(
```
Comment (garious) [inline src/private.rs]: Can you move this part into a separate function so that public.rs doesn't need to duplicate it?

**Comment by garious** [comment]
@iquerejeta, I see this PR has been open a long time. Any plans to move forward with it?

File: src/private.rs (hunk 1 above)
Comment (iquerejeta) [reply src/private.rs]: Yep, makes sense.

**Comment by iquerejeta** [comment]
We ended up not needing the functionality of the PR, but yes, I think I'll go forward with it. Looking again at the code, I'm thinking that it would make sense to make this function accept any hash function as input. WDYT?

**Comment by garious** [comment]
Sounds good to me. Or just use sha256 :)

**Comment by iquerejeta** [comment]
I'll merge, and think about generalising the hash function. We don't want the proof failing to verify due to a different choice of the hash function. Also, I tried to update the version sha2, and everything broke. I'll look into it a bit more tomorrow.

**Pull request merged**